packages/office-ui-fabric-react/src/components/Popup/Popup.tsx

```diff
@@ -65,20 +66,12 @@ export class Popup extends React.Component<IPopupProps, IPopupState> {
 
   public componentWillUnmount(): void {
     this._disposables.forEach((dispose: () => void) => dispose());
-    if (
-      this.props.shouldRestoreFocus &&
-      this._originalFocusedElement &&
-      this._containsFocus &&
-      (this._originalFocusedElement as any) !== window
-    ) {
-      // This slight delay is required so that we can unwind the stack, let react try to mess with focus, and then
-      // apply the correct focus. Without the setTimeout, we end up focusing the correct thing, and then React wants
-      // to reset the focus back to the thing it thinks should have been focused.
-      if (this._originalFocusedElement) {
-        this._originalFocusedElement.focus();
-      }
-    }
 
+    // tslint:disable-next-line: deprecation
+    if (this.props.shouldRestoreFocus) {
```

Comment:
any way to write a unit test to validate this works?

Reply:
Also a test to validate that focus is not restored if you provide an empty function.